src/postamble.js

```diff
@@ -484,15 +470,24 @@ if (!ENVIRONMENT_IS_PTHREAD) // EXIT_RUNTIME=0 only applies to default behavior
 #endif
 
 #if USE_PTHREADS
-if (!ENVIRONMENT_IS_PTHREAD) run();
+if (!ENVIRONMENT_IS_PTHREAD) {
+  run();
+} else {
 #if EMBIND
-else {  // Embind must initialize itself on all threads, as it generates support JS.
+  // Embind must initialize itself on all threads, as it generates support JS.
   Module['___embind_register_native_and_builtin_types']();
-}
 #endif // EMBIND
+#if MODULARIZE
+  // The promise resolve function typically gets called as part of the execution 
+  // of the Module `run`. The workers/pthreads don't execute `run` here, they
+  // call `run` in response to a message at a later time, so the creation
+  // promise can be resolved, marking the pthread-Module as initialized.
```

Comment:
why can't we wait for that later time? (that would be simpler if possible)

Reply:
If we don't resolve the promise here then the worker never sends the `loaded` message, and then the message to run the module never gets sent from the main thread.

Reply:
I agree this is a confusing flow. The minimal build makes this a little cleaner by having a `ready` function that gets called from the postamble.

Reply:
I see, sounds good.